api/handlers/refererhandler.py

```diff
@@ -0,0 +1,353 @@
+
+
+class AnalysesHandler(RefererHandler):
+    storage = containerstorage.AnalysisStorage()
+    storage_schema_file = 'analysis.json'
+    payload_schema_file = 'analysis.json'
+
+
+    def post(self, cont_name, cid):
+        """
+        Default behavior:
+            Creates an analysis object and uploads supplied input
+            and output files.
+        When param ``job`` is true:
+            Creates an analysis object and job object that reference
+            each other via ``job`` and ``destination`` fields. Job based
+            analyses are only allowed at the session level.
+        """
+        parent = self.storage.get_parent(cont_name, cid)
+        permchecker = self.get_permchecker(parent)
+        permchecker(noop)('POST')
+
+        if self.is_true('job'):
+            if cont_name != 'sessions':
+                self.abort(400, 'Analysis created via a job must be at the session level')
+
+            payload = self.request.json_body
+            analysis = payload.get('analysis')
+            job = payload.get('job')
+            if not analysis or not job:
+                self.abort(400, 'JSON body must contain map for "analysis" and "job"')
+            self.input_validator(analysis, 'POST')
+            result = self.storage.create_job_and_analysis(cont_name, cid, analysis, job, self.origin)
+            return {'_id': result['analysis']['_id']}
+
+        analysis = upload.process_upload(self.request, upload.Strategy.analysis, origin=self.origin)
+        self.storage.fill_values(analysis, cont_name, cid, self.origin)
+        result = self.storage.create_el(analysis)
+
+        if result.acknowledged:
+            return {'_id': result.inserted_id}
+        else:
+            self.abort(500, 'Analysis not added for container {} {}'.format(cont_name, cid))
+
+
+    def get(self, cont_name, cid, _id):
+        parent = self.storage.get_parent(cont_name, cid)
+        permchecker = self.get_permchecker(parent)
+        permchecker(noop)('GET')
+        return self.storage.get_container(_id)
+
+
+    @log_access(AccessType.delete_analysis)
+    def delete(self, cont_name, cid, _id):
+        parent = self.storage.get_parent(cont_name, cid)
+        permchecker = self.get_permchecker(parent)
+        permchecker(noop)('DELETE')
+        self.log_user_access(AccessType.delete_file, cont_name=cont_name, cont_id=cid)
+
+        try:
+            result = self.storage.delete_el(_id)
+        except APIStorageException as e:
+            self.abort(400, e.message)
+        if result.deleted_count == 1:
+            return {'deleted': result.deleted_count}
+        else:
+            self.abort(404, 'Analysis {} not removed from container {} {}'.format(_id, cont_name, cid))
+
+
+    def download(self, cont_name, cid, _id, filename=None):
+        """
+        .. http:get:: /api/(cont_name)/(cid)/analyses/(analysis_id)/files/(file_name)
+
+            Download a file from an analysis or download a tar of all files
```

Comment:
Feel free to trim this down. We no longer use Sphynx doc generation that this was written for.